libraries/Zigbee/examples/Zigbee_On_Off_Switch/Zigbee_On_Off_Switch.ino

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 Espressif Systems (Shanghai) PTE LTD
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @brief This example demonstrates simple Zigbee light switch.
+ *
+ * The example demonstrates how to use Zigbee library to control a light bulb.
+ * The light bulb is a Zigbee end device, which is controlled by a Zigbee coordinator (Switch).
+ * Button switch and Zigbee runs in separate tasks.
+ *
+ * Proper Zigbee mode must be selected in Tools->Zigbee mode
+ * and also the correct partition scheme must be selected in Tools->Partition Scheme.
+ *
+ * Please check the README.md for instructions and more detailed description.
+ * 
+ * Created by Jan Procházka (https://github.com/P-R-O-C-H-Y/)
+ */
+
+#ifndef ZIGBEE_MODE_ZCZR
+#error "Zigbee coordinator mode is not selected in Tools->Zigbee mode"
+#endif
+
+#include "ZigbeeCore.h"
+#include "ep/ZigbeeSwitch.h"
+
+#define SWITCH_ENDPOINT_NUMBER 5
+
+/* Switch configuration */
+#define GPIO_INPUT_IO_TOGGLE_SWITCH GPIO_NUM_9
+#define PAIR_SIZE(TYPE_STR_PAIR)    (sizeof(TYPE_STR_PAIR) / sizeof(TYPE_STR_PAIR[0]))
+
+typedef enum {
+  SWITCH_ON_CONTROL,
+  SWITCH_OFF_CONTROL,
+  SWITCH_ONOFF_TOGGLE_CONTROL,
+  SWITCH_LEVEL_UP_CONTROL,
+  SWITCH_LEVEL_DOWN_CONTROL,
+  SWITCH_LEVEL_CYCLE_CONTROL,
+  SWITCH_COLOR_CONTROL,
+} switch_func_t;
+
+typedef struct {
+  uint8_t pin;
+  switch_func_t func;
+} switch_func_pair_t;
+
+typedef enum {
+  SWITCH_IDLE,
+  SWITCH_PRESS_ARMED,
+  SWITCH_PRESS_DETECTED,
+  SWITCH_PRESSED,
+  SWITCH_RELEASE_DETECTED,
+} switch_state_t;
+
+static switch_func_pair_t button_func_pair[] = {{GPIO_INPUT_IO_TOGGLE_SWITCH, SWITCH_ONOFF_TOGGLE_CONTROL}};
+
+/* Zigbee switch */
+class MyZigbeeSwitch : public ZigbeeSwitch {
+public:
+    // Constructor that passes parameters to the base class constructor
+    MyZigbeeSwitch(uint8_t endpoint) : ZigbeeSwitch(endpoint) {}
+
+    // Override the set_on_off function
+    void readManufacturer(char* manufacturer) override {
+      //Do what you want with the manufacturer string
+      Serial.printf("Manufacturer: %s\n", manufacturer);
+    }
+    void readModel(char* model) override {
+      //Do what you want with the model string
+      Serial.printf("Model: %s\n", model);
+    }
+};
```

Comment:
is this class required here, if it's only to read manufacturer and model?

Reply:
It's not required there, I just wanted to have it implemented in some example so users can see how to use it if they want to. I can remove it if you prefer, as its not necessary for the example.